lib/solidus_content/content_providers/yaml.rb

```diff
@@ -1,13 +1,29 @@
+require 'yaml'
+
 module SolidusContent::ContentProviders::YAML
   def self.call(input)
     dir = Rails.root.join(input.dig(:type_options, :path))
     file = dir.join(input[:slug] + '.yml')
     file = dir.join(input[:slug] + '.yaml') unless file.exist?
 
-    data = YAML.load(file.read, symbolize_names: true, filename: file.to_s)
+    data = load(file)
 
     input.merge(data: data)
   end
 
+  # Ruby 2.6 deprecated passing the filename as the second argument, we'll do 
+  # a feature check instead of relying on the Ruby version.
+  filename_key = YAML.method(:load).parameters.rassoc(:filename).first == :key
+
+  if filename_key
```

Comment:
Why not this? I think that it is more understandable

```ruby
def self.load(file)
    return YAML.load(file.read, symbolize_names: true, filename: file.to_s) if filename_key

    YAML.load(file.read, file.to_s, symbolize_names: true)
 end
```

Reply:
This way we only do the check once instead of doing it each time the method is called, that's particularly apt since it depends on the language version, which is not likely to change  at runtime 😅